### ISSUE TITLE
Sometimes validation takes too much time/memory

### ISSUE THREAD

**Issue by blackbeam** [veraPDF/veraPDF-library#714]
I have relatively small document (40MB) on which validation (latest PDFBOX) takes ~ 55 minutes and ~ 22 GB of RAM. Maybe it is possible to  optimize?

I can send that file if needed.

Latest GF, in its turn, returns an error on this document:

```
org.verapdf.core.ValidationException: Caught unexpected runtime exception during validation
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:109)
	at org.verapdf.processor.ProcessorImpl.validate(ProcessorImpl.java:219)
	at org.verapdf.processor.ProcessorImpl.process(ProcessorImpl.java:120)
	at org.verapdf.processor.BatchFileProcessor.processItem(BatchFileProcessor.java:98)
	at org.verapdf.processor.BatchFileProcessor.processList(BatchFileProcessor.java:74)
	at org.verapdf.processor.AbstractBatchProcessor.process(AbstractBatchProcessor.java:102)
	at org.verapdf.cli.VeraPdfCliProcessor.processFiles(VeraPdfCliProcessor.java:155)
	at org.verapdf.cli.VeraPdfCliProcessor.processPaths(VeraPdfCliProcessor.java:129)
	at org.verapdf.cli.VeraPdfCli.main(VeraPdfCli.java:100)
Caused by: org.mozilla.javascript.WrappedException: Wrapped java.lang.NullPointerException (unnamed script#1)
	at org.mozilla.javascript.Context.throwAsScriptRuntimeEx(Context.java:1773)
	at org.mozilla.javascript.MemberBox.invoke(MemberBox.java:183)
	at org.mozilla.javascript.NativeJavaMethod.call(NativeJavaMethod.java:247)
	at org.mozilla.javascript.optimizer.OptRuntime.callProp0(OptRuntime.java:119)
	at org.mozilla.javascript.gen.c59._c0(unnamed script:1)
	at org.mozilla.javascript.gen.c59.call(unnamed script)
	at org.mozilla.javascript.ContextFactory.doTopCall(ContextFactory.java:398)
	at org.mozilla.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3065)
	at org.mozilla.javascript.gen.c59.call(unnamed script)
	at org.mozilla.javascript.gen.c59.exec(unnamed script)
	at org.verapdf.pdfa.validation.validators.BaseValidator.checkObjWithRule(BaseValidator.java:372)
	at org.verapdf.pdfa.validation.validators.BaseValidator.checkAllRules(BaseValidator.java:295)
	at org.verapdf.pdfa.validation.validators.BaseValidator.checkNext(BaseValidator.java:181)
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:136)
	at org.verapdf.pdfa.validation.validators.BaseValidator.validate(BaseValidator.java:107)
	... 8 more
Caused by: java.lang.NullPointerException
	at org.verapdf.pd.font.cmap.CMapFile.getDictWMode(CMapFile.java:51)
	at org.verapdf.gf.model.impl.external.GFCMapFile.getdictWMode(GFCMapFile.java:75)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mozilla.javascript.MemberBox.invoke(MemberBox.java:161)
	... 21 more
```

**Comment by bdoubrov** [comment]
Thanks for reporting the issue. Would you please provide a test file.

**Comment by bdoubrov** [comment]
The issues in GF version are fixed in 1.3.11: 
http://downloads.verapdf.org/dev/verapdf-1.3.11-GF-installer.zip

It is significantly faster and requires ~1Gb of memory.

**Comment by blackbeam** [comment]
Thanks!
